datasets/in/nse_debarred/crawler.py

```diff
@@ -0,0 +1,97 @@
+from typing import Generator
+from rigour.mime.types import XLS
+from normality import stringify, slugify
+from datetime import datetime
+import xlrd
+
+from zavod import Context, helpers as h
+
+SEBI_DEBARRMENT_URL = "https://nsearchives.nseindia.com/content/press/prs_ra_sebi.xls"
+OTHER_DEBARRMENT_URL = (
+    "https://nsearchives.nseindia.com/content/press/prs_ra_others.xls"
+)
+
+
+def parse_sheet(
+    sheet,
+) -> Generator[dict, None, None]:
+    headers = None
+    for row in sheet:
+        cells = []
+        for cell in row:
+            if cell.ctype == xlrd.XL_CELL_DATE:
+                # Convert Excel date format to Python datetime
+                date_value = xlrd.xldate_as_datetime(cell.value, sheet.book.datemode)
+                cells.append(date_value)
+            else:
+                cells.append(cell.value)
+        if headers is None:
+            headers = []
+            for idx, cell in enumerate(cells):
+                if not cell:
+                    cell = f"column_{idx}"
+                headers.append(slugify(cell, "_").lower())
+            continue
+
+        record = {}
+        for header, value in zip(headers, cells):
+            if isinstance(value, datetime):
+                value = value.date()
+            record[header] = stringify(value)
+        if len(record) == 0:
+            continue
+        yield record
+
+
+def crawl_item(input_dict: dict, context: Context):
+    name = input_dict.pop("entity_individual_name")
```

Comment:
let's split on:
`Proprietor of`, `Proprietor`
watch out, in the first case the right side is the company. In the second case the left side is the company. It looks like the PAN and the ban relate to the person, not the company.

So I think we want the company emitted as well, but you don't have to create a sanction for it. Just an ownership relation between the person and the company.

Reply:
Should I just do a simple split on those strings or use some kind of regex? Because there appears to be some variations:

- Mr. Naresh Pal (Proprietor of Equity Mantra)
- Mr. Shashikant Kumar (Owner of SM Global Research)
- M/s G N Traders (Proprietor – Mr. Kashif Siddique)
- M/s Ideal Equity (Proprietor: Anilkumar Talpada)
- GTR Solutions Financial Service(Prop.Nishant Chaturvedi)